aggregator/aggregator.go

```diff
@@ -158,7 +160,18 @@ func (a *Aggregator) processAndSubmitBatch(txs []core.Tx) {
 }
 
 func (a *Aggregator) processTxs(txs []core.Tx) (commitments []core.Commitment, err error) {
-	return db.ProcessTxs(&a.LoadedBazooka, &a.DB, txs, false)
+	var txsInCommitment []int
+	if len(txs)%int(config.GlobalCfg.TxsPerCommitment) != 0 {
+		return commitments, ErrIncorrectTxCount
+	}
+
+	for i := range txs {
+		if i%int(config.GlobalCfg.TxsPerCommitment) == 0 {
+			txsInCommitment = append(txsInCommitment, int(config.GlobalCfg.TxsPerCommitment))
+		}
+	}
+
+	return db.ProcessTxs(&a.LoadedBazooka, &a.DB, txs, txsInCommitment, false)
```

Comment:
What are our assumptions here?

Note that 

1. Only the same transaction type can be committed together.
2. Some commitments might be only partially filled.

Reply:
Yeah here we assume we won't partially fill a commitment, the PopTxs() function takes care of only pulling in the right transactions from the pool